.github/workflows/yocto-build-deploy.yml

```
@@ -839,10 +846,8 @@ jobs:
             fi
           done
 
-      # Note: in the original "yocto-scripts" there were a few checks to ensure that the release was a finalised version, and that it didn't already have a version tag
-      # The versions tags are legacy now anyway - so I haven't included that - and we know this will be a finalised release anyway
       - name: Tag ESR release
-        if: inputs.deploy-hostapp == true && inputs.deploy-esr && (steps.merge-test-result.outputs.finalize || inputs.finalize-hostapp)
+        if: steps.should-deploy.outputs.deploy == true && steps.esr-check.outputs.is-esr && steps.should-finalize.outputs.finalize == true
```

```suggestion
        if: steps.should-deploy.outputs.deploy == 'true' && steps.esr-check.outputs.is-esr == 'true' && steps.should-finalize.outputs.finalize == 'true'
```
These outputs are all strings as I pointed out above, but there are other ways to evaluate these steps by looking at the step outcome.